tests/gitlab/build_and_test

```diff
@@ -130,6 +137,19 @@ timed_message ()
     echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
 }
 
+at_exit()
+{
+  if [[ "${#exit_rm_list[@]}" != 0 ]]; then
+    echo ${script_name}:at_exit: rm -rf "${exit_rm_list[@]}"
+    rm -rf "${exit_rm_list[@]}"
+  fi
+  if [[ "${exit_distclean}" == 1 ]]; then
+    echo ${script_name}:at_exit: \
+      make -C "${project_dir}" distclean "> /dev/null 2>&1"
+    make -C "${project_dir}" distclean > /dev/null 2>&1
+  fi
+}
+
```

Comment:
I think this should be conditional to _not running in debug mode_.

Reply:
Okay, I'll add a check for debug mode and skip all the cleanup if debug mode is enabled.